src/utils/codapPhone/types.ts

```diff
@@ -0,0 +1,95 @@
+export enum CodapComponent {
+  Graph = "graph",
+  Table = "caseTable",
+  Map = "map",
+}
+
+export enum CodapResource {
+  DataContext = "dataContext",
+  DataContextList = "dataContextList",
+  Component = "component",
+}
+
+export enum CodapActions {
+  Create = "create",
+  Update = "update",
+  Delete = "delete",
+  Get = "get",
+  Notify = "notify",
+}
+
+export type CodapRequest = {
+  action: CodapActions;
+  resource: string;
+  values?: any;
+}
+
+export interface CodapResponse {
+  success: boolean;
+}
+
+export interface CodapResponseValues extends CodapResponse {
+  values?: any;
+}
```

Comment:
I think we may eventually want to fully annotate the structure of all codap objects (ex: what fields can a data context have, what fields can a collection have, etc.) which will let us make this type more concrete.